src/reporter.cc

```diff
@@ -144,20 +144,30 @@ void MetadataReporter::SendMetadata(
 
   std::vector<json::value> entries;
   for (auto& entry : metadata) {
-    const MonitoredResource& resource = entry.first;
+    const std::string& full_resource_name = entry.first;
     MetadataStore::Metadata& metadata = entry.second;
     if (metadata.ignore) {
       continue;
     }
     json::value metadata_entry =
-        json::object({  // MonitoredResourceMetadata
-          {"resource", resource.ToJSON()},
-          {"rawContentVersion", json::string(metadata.version)},
-          {"rawContent", std::move(metadata.metadata)},
-          {"state", json::string(metadata.is_deleted ? "DELETED" : "ACTIVE")},
-          {"createTime", json::string(time::rfc3339::ToString(metadata.created_at))},
-          {"collectTime", json::string(time::rfc3339::ToString(metadata.collected_at))},
-        });
+        json::object({
+          {"name", json::string(full_resource_name)},
+          {"type", json::string(metadata.type)},
+          {"location", json::string(metadata.location)},
+          {"state", json::string(metadata.is_deleted ? "DELETED" : "EXISTS")},
+          {"eventTime", json::string(
+              time::rfc3339::ToString(metadata.collected_at))},
+          {"views",
+            json::object({
+              {metadata.version,
+                json::object({
+                  {"schemaName", json::string(metadata.schema_name)},
+                  {"stringContent", json::string(metadata.metadata->ToString())},
+                })
+              }
+            })
+          }
+      });
     // TODO: This is probably all kinds of inefficient...
     const int size = metadata_entry->ToString().size();
```

Comment:
I don't believe this code applies anymore — each metadata entry is its own request in the batch endpoint, so computing the total size is nonsensical.

Reply:
We still want to control the size of the batch request right?
In any case, this PR does not update the actual request - that will come later.